compiler/rustc_hir_analysis/src/lib.rs

```diff
@@ -196,6 +196,15 @@ pub fn check_crate(tcx: TyCtxt<'_>) -> Result<(), ErrorGuaranteed> {
         collect::test_opaque_hidden_types(tcx)?;
     }
 
+    tcx.hir().par_body_owners(|item_def_id| {
```

Comment:
```suggestion
    // Make sure we evaluate all static and (non-associated) const items, even if unused.
    // If any of these fail to evaluate, we do not want this crate to pass compilation.
    tcx.hir().par_body_owners(|item_def_id| {
```
And maybe also say something about the DefId generation, which forces this to be here? I didn't understand that part.

Reply:
That is a comment I already rebased away locally 😅

Reply:
I... don't know what that means.^^

Reply:
> And maybe also say something about the DefId generation

that's not happening. We're not generating def ids yet. I will do so in the future and then update the comment here